### DISCUSSION TITLE
Different highlight colors for member and local variables

### DISCUSSION THREAD

**Discussion by TweedleDim** [godotengine/godot-proposals#4605] [Editor]
In the Editor Settings /Text Editor/Highlighting there is a _Member Variable Color_
**BUT** a member variable will only be displayed with this colour when used with the point notation self.my_variable

Otherwise it appears with the _Text color,_ as well as many other stuff, such as local variables.

Wouldn't you find it useful, as I do, to be able to easily distinguish between member and local variables in your code, with 2 clearly different tones ?

Thumbs up if you would ^_^

**Comment by Mickeon** [comment]
Always found it odd. I considered opening a proposal for having declared properties in the same script have the same colour as other in-engine properties (`position`, `visible`, etc. etc.) or ones preceded by `self`, it's what I would expect to be the case, even if I can see why someone may want the current behavior.

**Comment by Calinou** [comment]
I think this would be a useful addition. However, for this feature to be implemented properly, this requires making the GDScript syntax highlighter able to semantically understand the code it's highlighting. This is a lot of work to do correctly.

Even highlighting constants separately from variables can be surprisingly nontrivial given how the syntax highlighter currently works.

**Comment by TweedleDim** [comment]
Now that you mention it, having constants with a colour of their own would be nice too !
I guess this is indeed something harder to implement than one would think but I hope these new highlightings could be added one day.

**Reply by Mickeon** [reply]
Sounds nice, although in practice, the Script Editor already features a lot of colours, each symbolising their own thing. Also, oftentimes, because the standard is to write constants in UPPER_CASE, or at least in a way that's recognisable by itself, a special colour for them too may not be top-priority.
